### PR TITLE
Fixes dependency version format error

### PR DESCRIPTION
**Pull request by mario-eth**
## Motivation

Fixing https://github.com/mario-eth/soldeer/issues/87

When the version is a string, the forge will return an error for incorrect config mapping.

## Solution
Using a custom Enum that can map to a string or to a map.

### CODE REVIEW

File: crates/config/src/soldeer.rs [hunk 1]
```diff
@@ -17,10 +17,17 @@ pub struct SoldeerDependency {
...
 
 impl AsRef<Self> for SoldeerConfig {
     fn as_ref(&self) -> &Self {
         self
     }
 }
+
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum SoldeerDependencyValue {
```
Comment (DaniPopes) [inline crates/config/src/soldeer.rs:30]: #[serde(untagged)]?

Comment (mario-eth) [reply crates/config/src/soldeer.rs:30]: should i add that?

Comment (mario-eth) [reply crates/config/src/soldeer.rs:31]: added

File: crates/config/src/soldeer.rs [hunk 2]
```diff
@@ -17,10 +17,17 @@ pub struct SoldeerDependency {
...
 
 impl AsRef<Self> for SoldeerConfig {
     fn as_ref(&self) -> &Self {
         self
     }
 }
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum SoldeerDependencyValue {
```
Comment (mattsse) [inline crates/config/src/soldeer.rs:30]: could we get some basic docs here, onliners

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**